### DISCUSSION TITLE
[Bug]: Deleting a conversation while waiting for a generation in another conversation delete both

### DISCUSSION THREAD

**Discussion by ThomasBaruzier** [danny-avila/LibreChat#3859] [Troubleshooting]
### What happened?

Hi, i would like report a bug:

Deleting a conversation while waiting for a generation in another conversation delete both

### Steps to Reproduce

1. Start a conversation
2. Start a new one, send a large prompt (in order to have the time for next step)
3. Delete the first converation while the second one waits for the first response token
4. Both conversations are deleted

### Screenshots

[screen2.webm](https://github.com/user-attachments/assets/bdbaf5d9-a16d-47c7-b09b-818e703fa589)

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
I can't reproduce this one:

https://github.com/user-attachments/assets/aef81a03-d93b-4891-9416-89d5122d31c8

**Comment by danny-avila** [comment]
Also, is the second conversation there after a refresh? It could be a client-side caching issue

**Comment by ThomasBaruzier** [comment]
@danny-avila Reloading the page does not bring back the deleted conversation that wasn't supposed to be deleted. I used one client.

Also, to reproduce the bug, you need to delete the previous conversation before the new one generates the first token, not while it is generating. So you might need an endpoint that takes a lot of time to perform prompt ingestion, such as a local one, or a very long input prompt with a large frontier model, but that costs a bit of money for a simple test.

**Reply by danny-avila** [reply]
Oh I see, well in that case, the second conversation has not even been created yet, so the UI kind of reflects that?

I agree that the UX should be better, though an edge case, it must currently mess with the conversation state.

**Reply by ThomasBaruzier** [reply]
Yes, probably. But a conversation waiting for its first token shoudn't be deleted upon the deletion of another conversation. It's not very important since it's an edge case which can be easily avoidable, but I still lost a prompt I had a hard time finding again. Is it what it is 😅

Edit: Could creating the conversation as soon as a prompt is sent be a solution?

**Reply by ThomasBaruzier** [reply]
I just got bamboozled with the same issue, but this time, it was more subtle.

I crafted a very long prompt, maybe over 128k tokens, and got an error from the API. I refresh the page, the conversation is gone ._.

**Reply by ThomasBaruzier** [reply]
Here's a demo:

[output.webm](https://github.com/user-attachments/assets/85dba2f9-ea46-43a0-8dd6-019a618f21df)

- I create a new conversation
- Paste and send a prompt
- The API fails
- When I click on the current conversation or refresh the page, it vanishes
